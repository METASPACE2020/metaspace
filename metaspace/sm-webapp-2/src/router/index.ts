import { createRouter, createWebHistory } from 'vue-router'
import { Component } from 'vue'
import NotFound from '../modules/App/NotFoundPage.vue'
import AboutPage from '../modules/App/AboutPage'
import DatasetsPage from '../modules/Datasets/DatasetsPage.vue'

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ '../modules/Annotations/AnnotationsPage.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ '../modules/MetadataEditor/MetadataEditPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ '../modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/imzml/DatasetBrowserPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/HelpPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Project/ProjectsListPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/PrivacyPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Project/ViewProjectPage.vue'),

}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, Component>

export const routes : any =[
  { path: '/', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/about', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
  {
    path: '/datasets',
    component: DatasetsPage,
    children: [
      { path: '', component: asyncPages.DatasetTable },
      { path: 'summary', component: asyncPages.DatasetTable },
    ],
  },
  {
    path: '/datasets/:dataset_id/comparison',
    name: 'datasets-comparison',
    component: asyncPages.DatasetComparisonPage,
  },
  { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
  { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.DatasetTable },
  { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
  { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
  { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
  { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetTable },
  { path: '/upload', component: asyncPages.UploadPage },
  { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },



  { path: '/groups', component: asyncPages.GroupsListPage },
  { path: '/group/create', component: asyncPages.CreateGroupPage },
  { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },


  { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
  { // Legacy URL sent in "request access" emails up until Feb 2019
    path: '/project/:projectIdOrSlug/manage',
    redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } },
  },
  { path: '/projects', component: asyncPages.ProjectsListPage },



  { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },


  {
    path: '/:pathMatch(.*)*',
    component: NotFound,
  },
]

const router = createRouter({
  // @ts-ignore
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior(to, from, savedPosition) {
    // return desired position
    if (savedPosition) {
      return savedPosition;
    }
    return { top: 0 };
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})


const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
