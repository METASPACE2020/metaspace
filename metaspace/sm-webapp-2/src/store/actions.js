import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "vue";
import store from "../store/index";
import reportError from "@/lib/reportError";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');
    let response = {}

    try{
      response = await apolloClient.query({
        query: fetchOptionListsQuery,
        fetchPolicy: 'cache-first',
      });
    } catch (err) {
      reportError(err);
    }

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
