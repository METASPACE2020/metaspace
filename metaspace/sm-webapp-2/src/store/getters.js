export default {
  filter() {
    return {}
    // return decodeParams(state.route, state.filterLists);
  },

  currentUser(state) {
    return state.currentUser;
  },
}
