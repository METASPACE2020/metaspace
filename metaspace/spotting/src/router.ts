/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter from 'vue-router'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  DashboardPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Dashboard/DashboardPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(dashboard)?', name: 'dashboard', component: asyncPages.DashboardPage },
    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
