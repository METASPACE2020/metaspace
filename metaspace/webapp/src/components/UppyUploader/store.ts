// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
