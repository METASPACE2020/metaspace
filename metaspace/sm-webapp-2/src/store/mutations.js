import { isEqual, omit, without } from 'lodash-es'
import router from '../router'
import compare from '../lib/compare'
import {
  decodeParams,
  encodeParams,
  encodeSections,
  encodeSortOrder,
  FILTER_SPECIFICATIONS,
  getFilterInitialValue,
  stripFilteringParams,
} from '../modules/Filters'
import { DEFAULT_ANNOTATION_VIEW_SECTIONS, DEFAULT_COLORMAP, DEFAULT_TABLE_ORDER } from '../modules/Filters/url'
import { DEFAULT_SCALE_TYPE } from '../lib/constants'

function updatedLocation(state, filter) {
  const query = encodeParams(filter, state.route.path, state.filterLists)

  state.lastUsedFilters[state.route.path] = {
    filter,
    query,
    order: state.orderedActiveFilters,
  }

  return {
    query: Object.assign(query, stripFilteringParams(state.route.query)),
  }
}

function routerReplace(state, params) {
  router.replace(params)
  state.route.path = params.path || state.route.path
  state.route.params = params.params || state.route.params
  state.route.query = params.query || state.route.query
}

function routerPush(state, params) {
  router.push(params)
  state.route.path = params.path || state.route.path
  state.route.params = params.params || state.route.params
  state.route.query = params.query || state.route.query
}

function replaceURL(state, filter) {
  routerReplace(state, updatedLocation(state, filter))
}

function pushURL(state, filter) {
  routerPush(state, updatedLocation(state, filter))
}

function sortFilterKeys(keys) {
  keys.sort((a, b) => compare(FILTER_SPECIFICATIONS[a].sortOrder || 100, FILTER_SPECIFICATIONS[b].sortOrder || 100))
}

function updateFilter(state, filter, routerAction = null) {
  const active = []

  // drop unset filters
  for (let i = 0; i < state.orderedActiveFilters.length; i++) {
    const key = state.orderedActiveFilters[i]
    if (filter[key] !== undefined) active.push(key)
  }

  // append newly added filters to the end
  for (const key in filter) if (filter[key] !== undefined && active.indexOf(key) == -1) active.push(key)

  // sort
  sortFilterKeys(active)

  const changedFilterSet = !isEqual(state.orderedActiveFilters, active)

  state.orderedActiveFilters = active
  if (routerAction === 'push' || (routerAction == null && changedFilterSet)) pushURL(state, filter)
  else replaceURL(state, filter)
}

export default {
  updateRoute(state, route) {
    state.route.path = route.path
    state.route.params = route.params
    state.route.query = route.query
  },

  updateFilter(state, filter) {
    // TODO: Stop using updateFilter for adding/removing filters, as it makes reacting to added/removed filters
    // much more difficult
    updateFilter(state, filter)
  },

  updateCurrentUser(state, user) {
    state.currentUser = user
  },

  setReportError(state) {
    state.reportError = !state.reportError
  },

  replaceFilter(state, filter) {
    updateFilter(state, filter, 'replace')
  },

  addFilter(state, name) {
    const oldFilter = decodeParams(state.route, state.filterLists)
    const filtersToAdd = [name]
    const filtersToRemove = []

    // Check for required additional filters & conflicting filters
    ;(FILTER_SPECIFICATIONS[name].dependsOnFilters || []).forEach((key) => {
      if (oldFilter[key] === undefined) {
        filtersToAdd.push(key)
      }
    })
    ;(FILTER_SPECIFICATIONS[name].conflictsWithFilters || []).forEach((key) => {
      if (oldFilter[key] !== undefined) {
        filtersToRemove.push(key)
      }
    })

    const newFilter = { ...oldFilter }
    filtersToAdd.forEach((key) => {
      newFilter[key] = getFilterInitialValue(key, state.filterLists)
    })
    filtersToRemove.forEach((key) => {
      newFilter[key] = undefined
    })
    const newActive = without([...state.orderedActiveFilters, ...filtersToAdd], ...filtersToRemove)
    sortFilterKeys(newActive)

    state.orderedActiveFilters = newActive
    pushURL(state, newFilter)
  },

  removeFilter(state, name) {
    const oldFilter = decodeParams(state.route, state.filterLists)
    const filtersToRemove = [name]

    // Check for dependent filters that should also be removed
    Object.keys(oldFilter).forEach((key) => {
      const { dependsOnFilters } = FILTER_SPECIFICATIONS[key]
      if (dependsOnFilters != null && dependsOnFilters.includes(name)) {
        filtersToRemove.push(key)
      }
    })

    const newFilter = omit(oldFilter, filtersToRemove)
    state.orderedActiveFilters = without(state.orderedActiveFilters, ...filtersToRemove)
    pushURL(state, newFilter)
  },

  updateFilterOnNavigate(state, to) {
    const newActiveFilters = Object.keys(decodeParams(to, state.filterLists))

    const removed = without(state.orderedActiveFilters, ...newActiveFilters)
    const added = without(newActiveFilters, ...state.orderedActiveFilters)

    state.orderedActiveFilters = without(state.orderedActiveFilters, ...removed)
    state.orderedActiveFilters.push(...added)
    sortFilterKeys(state.orderedActiveFilters)
  },

  setFilterListsLoading(state) {
    state.filterListsLoading = true
  },

  setFilterLists(state, filterLists) {
    state.filterLists = filterLists
    state.filterListsLoading = false
  },

  setAnnotation(state, annotation) {
    state.annotation = annotation
  },

  setNormalizationMatrix(state, normalizationMatrix) {
    state.normalization = normalizationMatrix
  },

  resetChannels(state) {
    state.channels = []
  },

  addChannel(state, { id, annotations, settings }) {
    state.channels.push({ id, annotations, settings })
  },

  restoreChannels(state, channels) {
    state.channels = channels
  },

  removeChannel(state, { index }) {
    const aux = state.channels.slice(0)
    aux.splice(index, 1)
    state.channels = aux
  },

  updateChannel(state, { index, id, annotations, settings }) {
    const aux = state.channels.slice(0)
    aux[index] = { id, annotations, settings }
    state.channels = aux
  },

  setViewerMode(state, mode = 'SINGLE') {
    state.mode = mode
  },

  setRoiInfo(state, { key, roi }) {
    state.roiInfo[key] = roi
  },

  toggleRoiVisibility(state, visible) {
    state.roiInfo['visible'] = visible
  },

  resetRoiInfo(state) {
    state.roiInfo = { visible: false }
  },

  setSnapshotAnnotationIds(state, annotation) {
    state.snapshotAnnotationIds = annotation
  },

  updateAnnotationTableStatus(state, isLoading) {
    state.tableIsLoading = isLoading
  },

  startTour(state, tourDescription) {
    state.currentTour = tourDescription
  },

  endTour(state) {
    state.currentTour = null
  },

  updateAnnotationViewSections(state, activeSections) {
    const sections = encodeSections(activeSections)
    const defaultSections = encodeSections(DEFAULT_ANNOTATION_VIEW_SECTIONS)
    routerReplace(state, {
      query: sections !== defaultSections ? { ...state.route.query, sections } : omit(state.route.query, 'sections'),
    })
  },

  setColormap(state, cmap) {
    routerReplace(state, {
      query: cmap !== DEFAULT_COLORMAP ? { ...state.route.query, cmap } : omit(state.route.query, 'cmap'),
    })
  },

  setScaleType(state, scaleType) {
    routerReplace(state, {
      query:
        scaleType !== DEFAULT_SCALE_TYPE
          ? { ...state.route.query, scale: scaleType }
          : omit(state.route.query, 'scale'),
    })
  },

  setLockTemplate(state, lockTemplate) {
    routerReplace(state, {
      query: lockTemplate ? { ...state.route.query, lock: lockTemplate } : omit(state.route.query, 'lock'),
    })
  },

  setNormalization(state, normalization) {
    routerReplace(state, {
      query: !normalization ? omit(state.route.query, 'norm') : { ...state.route.query, norm: normalization },
    })
  },

  setCurrentPage(state, page) {
    routerReplace(state, {
      query: page !== 1 ? { ...state.route.query, page: String(page) } : omit(state.route.query, 'page'),
    })
  },

  setRow(state, row) {
    routerReplace(state, {
      query: row !== 1 ? { ...state.route.query, row: String(row) } : omit(state.route.query, 'row'),
    })
  },

  setSortOrder(state, sortOrder) {
    const sort = encodeSortOrder(sortOrder)
    const defaultSort = encodeSortOrder(DEFAULT_TABLE_ORDER)
    routerReplace(state, {
      query: sort !== defaultSort ? { ...state.route.query, sort } : omit(state.route.query, 'sort'),
    })
  },

  setColocalizationAlgo(state, colocalizationAlgo) {
    routerReplace(state, {
      query:
        colocalizationAlgo != null ? { ...state.route.query, alg: colocalizationAlgo } : omit(state.route.query, 'alg'),
    })
  },

  setDatasetTab(state, tab) {
    const path = {
      list: '/datasets',
      summary: '/datasets/summary',
    }[tab.toLowerCase()]
    routerReplace(state, {
      path,
      query: state.route.query,
    })
  },
}
