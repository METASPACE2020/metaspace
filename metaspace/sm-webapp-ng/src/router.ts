import { createRouter, createWebHistory } from 'vue-router'
import HomeView from './views/HomeView.vue'
import AboutView from './views/AboutView.vue'
import store from './store'
// ... other imports ...

const routes = [
  {
    path: '/',
    component: HomeView
  },
  {
    path: '/user/:id',
    component: AboutView
  }
]

const router = createRouter({
  history: createWebHistory(),
  routes
})

router.afterEach((to) => {
  console.log('to', to)
  // store.commit('updateFilterOnNavigate', to)
})

export default router
