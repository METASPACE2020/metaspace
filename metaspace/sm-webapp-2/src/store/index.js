import { createStore } from 'vuex'
import mutations from './mutations'
import getters from './getters'
import actions from './actions.js'

import { accountModule } from '../modules/Account'

const store = createStore({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    reportError: false,

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: { visible: false },

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE',
    route: {
      path: '',
      params: {},
      query: {},
    },
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule,
  },
})

export default store
