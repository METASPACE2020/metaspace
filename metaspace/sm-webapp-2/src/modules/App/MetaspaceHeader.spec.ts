import { describe, it, expect, vi } from 'vitest';
import router from "@/router";
import store from "@/store";
import { nextTick, ref } from 'vue';
import {flushPromises, mount} from '@vue/test-utils';
import MyComponent from './MetaspaceHeader';

const currentUserMockResponses = [
  {
    data: {
      currentUser:  {
        id: '123',
        name: 'Test UserX',
        email: 'test@example.com',
        primaryGroup: {
          group: {
            id: '456',
            name: 'Test Group',
            urlSlug: null,
          },
        },
        groups: [
          { role: 'MEMBER', group: { hasPendingRequest: null } },
          { role: 'GROUP_ADMIN', group: { hasPendingRequest: false } },
        ],
        projects: [
          { role: 'PENDING', project: { hasPendingRequest: null } },
          { role: 'MANAGER', project: { hasPendingRequest: true } },
        ],
      },
    },
  },
  {
    data: {
      currentUser: null,
    },
  }
];

let mockIndex = 0; // This will determine which mock response to use

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(() => ({
    result: ref(currentUserMockResponses[mockIndex].data),
    loading: ref(false),
    error: ref(null),
    subscribeToMore: vi.fn(),
    // Implement any other methods that your component expects
  })),
  // ...mock other composable if needed
}));


// Mock router's push function or any other function you need
router.push = vi.fn();


describe('MyComponent', () => {
  it('renders with first user data', async () => {
    mockIndex = 0;
    // Mount the component using Vue Test Utils
    const wrapper = mount(MyComponent, {
      global: {
        plugins: [router, store], // Use the mock router as a plugin
      },
    });

    await nextTick(); // Wait for the next "tick" after mounting

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with second user data', async () => {
    mockIndex = 1;
    // Mount the component using Vue Test Utils
    const wrapper = mount(MyComponent, {
      global: {
        plugins: [router, store], // Use the mock router as a plugin
      },
    });

    await nextTick(); // Wait for the next "tick" after mounting

    expect(wrapper.html()).toMatchSnapshot();
  });


  afterEach(() => {
    // Reset the mockIndex or any other state between tests
    mockIndex = 0;
    vi.clearAllMocks();
  });
});
