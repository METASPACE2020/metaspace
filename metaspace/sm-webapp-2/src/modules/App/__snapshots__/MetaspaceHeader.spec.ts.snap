// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyComponent > renders with first user data 1`] = `
"<div class=\\"sm-header h-16\\">
  <div class=\\"fixed top-0 left-0 right-0\\">
    <div class=\\"
              non-responsive-menu
              transition-colors duration-300 ease-in-out h-16
              flex items-center justify-between
              bg-primary\\">
      <div class=\\"header-items\\"><a href=\\"/\\" class=\\"flex pl-3 pr-4\\"><img src=\\"/src/assets/images/logo.png\\" alt=\\"Metaspace\\" title=\\"Metaspace\\"></a><a href=\\"/upload\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"upload-link\\">Upload</a>,<a href=\\"/annotations\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"annotations-link\\">Annotations</a>,<a href=\\"/datasets\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"datasets-link\\">Datasets</a>,<a href=\\"/projects\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Projects</a>,<a href=\\"/groups\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Groups</a></div>
      <div class=\\"header-items\\"><a href=\\"/detectability\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Detectability</a>,<a href=\\"/help\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Help</a>
        <!---->
        <div class=\\"header-items mr-1 lg:mr-2\\">
          <div class=\\"relative flex py-2\\"><a href=\\"/user/me\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent\\" id=\\"user-menu\\">Test UserX<span data-v-6bf8fbbe=\\"\\" class=\\"notification el-tooltip__trigger el-tooltip__trigger\\"></span>
              <!--teleport start-->
              <!--teleport end-->
            </a>
            <transition-stub enterfromclass=\\"transform opacity-0 scale-95\\" entertoclass=\\"transform opacity-100 scale-100\\" leavefromclass=\\"transform opacity-100 scale-100\\" leavetoclass=\\"transform opacity-0 scale-95\\" enteractiveclass=\\"transition ease-out duration-100\\" leaveactiveclass=\\"transition ease-in duration-75\\"></transition-stub>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"responsive-menu transition-colors duration-300 ease-in-out h-16 flex items-start justify-center bg-primary flex-wrap h-full\\">
      <div class=\\"header-items flex-row w-full justify-between\\" style=\\"height: 64px;\\"><a href=\\"/\\" class=\\"flex pl-3 pr-4\\"><img src=\\"/src/assets/images/logo.png\\" alt=\\"Metaspace\\" title=\\"Metaspace\\"></a><button class=\\" button-reset flex h-12 w-12 mr-3\\"><svg class=\\"h-full w-full sm-menu-icon\\"></svg>
          <!---->
        </button></div>
      <!---->
    </div>
    <!---->
  </div>
</div>"
`;

exports[`MyComponent > renders with second user data 1`] = `
"<div class=\\"sm-header h-16\\">
  <div class=\\"fixed top-0 left-0 right-0\\">
    <div class=\\"
              non-responsive-menu
              transition-colors duration-300 ease-in-out h-16
              flex items-center justify-between
              bg-primary\\">
      <div class=\\"header-items\\"><a href=\\"/\\" class=\\"flex pl-3 pr-4\\"><img src=\\"/src/assets/images/logo.png\\" alt=\\"Metaspace\\" title=\\"Metaspace\\"></a><a href=\\"/upload\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"upload-link\\">Upload</a>,<a href=\\"/annotations\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"annotations-link\\">Annotations</a>,<a href=\\"/datasets\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\" id=\\"datasets-link\\">Datasets</a>,<a href=\\"/projects\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Projects</a>,<a href=\\"/groups\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Groups</a></div>
      <div class=\\"header-items\\"><a href=\\"/detectability\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Detectability</a>,<a href=\\"/help\\" class=\\"mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Help</a>
        <div class=\\"header-items mr-1 lg:mr-2\\"><button class=\\"button-reset 
  mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Create account</button>,<button class=\\"button-reset 
  mx-1 px-3 py-2 rounded-md no-underline
  text-sm font-medium text-white lg:tracking-wide
  hover:bg-blue-700 focus:outline-none focus:bg-blue-700
  transition-colors duration-150 ease-in-out
  border border-solid border-transparent header-link\\">Sign in</button></div>
        <!---->
      </div>
    </div>
    <div class=\\"responsive-menu transition-colors duration-300 ease-in-out h-16 flex items-start justify-center bg-primary flex-wrap h-full\\">
      <div class=\\"header-items flex-row w-full justify-between\\" style=\\"height: 64px;\\"><a href=\\"/\\" class=\\"flex pl-3 pr-4\\"><img src=\\"/src/assets/images/logo.png\\" alt=\\"Metaspace\\" title=\\"Metaspace\\"></a><button class=\\" button-reset flex h-12 w-12 mr-3\\"><svg class=\\"h-full w-full sm-menu-icon\\"></svg>
          <!---->
        </button></div>
      <!---->
    </div>
    <!---->
  </div>
</div>"
`;
